### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: nested-functions

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1717]
This issue describes how to implement the `nested-functions` concept exercise for the <LANG> track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Learning objectives

- Be able to define local / nested functions using `flet` and `labels`
- Know that, like lexical variables, these functions can shadow global ones
- Understand that `flet`-defined functions aren't able to call each other or themselves (like `let`)
- Understand that `labels` allows for functions that are aware of each other (similar to `let*`)

## Out of scope

- Recursion (This is a prerequisite for recursion)
- The difference between special operators and functions
- Macros and `macrolet`

## Concepts

- `nested-functions`

## Prerequisites

- `variables`
- `function-definition`
- `higher-order-functions`
- `symbols`

## Resources to refer to

[Special Operators – Practical Common Lisp](http://www.gigamonkeys.com/book/the-special-operators.html)
[CLHS](http://clhs.lisp.se/Body/s_flet_.htm)

### Hints

Nothing yet

### After

- [Code Style Considerations](https://softwareengineering.stackexchange.com/questions/363084/use-labels-or-nested-flet)

## Representer

- Be sure to normalise the names of local functions

## Analyzer

- Be able to ensure that local functions are being used as opposed to global ones (or lambdas)

## Implementing

While `labels` is often used for recursion, avoid that here, as `recursion` comes after this exercise.

Many students may wonder why they would ever want to use `flet` when the "more powerful" `labels` exists. To this end, it might be good to include an example of `flet` "wrapping" some global function with the same name, like:
```lisp
(defun my-car (lst)
  (car lst))

(defun get-cars (alist)
  (flet ((my-car (lst) (print "Fetching a car!") (my-car lst)))
    (mapcar #'my-car alist)))

(get-cars (pairlis '(#\a #\b #\c) '(1 2 3)))
; => "Fetching a car!"
; => "Fetching a car!"
; => "Fetching a car!"
; => (#\c #\b #\a)
```

That function shadowing / wrapping wouldn't be possible with `labels` as the rebinding of `my-car` would appear recursive, while with `flet`, `(my-car lst)` refers to the outer binding of `my-car`.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
